### ISSUE TITLE
phpMyAdmin 5.2.1 error afer login on Ubuntu focal: `Unknown named parameter $$dbi`

### ISSUE THREAD

**Issue by Danieleeffe1** [phpmyadmin/phpmyadmin#18498]
### Please describe the bug

After logging in, a page appears that says:
  Error: Unknown named parameter $$dbi

### To be reproduced

I add ppa
update ppa
software upgrades
install new version 5.2.1 from 4.9x
I select "Y" to the question "replace old parameters with new ones"
system restart

### Expected behavior

Ubuntu server 20.04 update PHPMYADMIN

### Server configuration

- Operating system: ubuntu server 20.04
- Webserver: 8.0.33-0ubuntu0.20.04.2
- Database version: MySql
- PHP version: 7.4
- phpMyAdmin version: 5.2.1

### Client configuration

-Browser: all
- Operating system: all

### Additional context

-I go to the URL address www.myserver.xxx/phpmyadmin
- the login window appears
-I enter my login details
-a blank page appears with this wording: Error: Unknown named parameter $$dbi

I tried without solving:
uninstall/clean system and reinstall phpmyadmin

**Comment by ibennetch** [comment]
I believe the problem you're seeing can be solved by running PHP 8 instead of 7.4.

You can read quite a bit about the matter at https://github.com/phpmyadmin/phpmyadmin/wiki/DebianUbuntu

**Comment by williamdes** [comment]
> I believe the problem you're seeing can be solved by running PHP 8 instead of 7.4.

Yes, you need PHP 8, for the reasons explained here: https://github.com/phpmyadmin/phpmyadmin/wiki/DebianUbuntu#why-does-the-ppa-only-support-php-80

Can you send the output of `apt policy phpmyadmin` ?

**Comment by Danieleeffe1** [comment]
thanks for the replies

premise:

i upgraded on two ubuntu 20.04 servers
both servers have multiple php versions installed
both use php 7.4

server A has php 5.6, 7.2, 7.4 installed and by default it uses php 7.4 and it works myphpadmin 5.2.1

Server B has php7.2, 7.4, 8.1, 8.2 installed and by default it uses php7.4 and myphpadmin 5.2.1 doesn't work

in server B I have PHP 8.1 enabled by default, result:
myphpadmin doesn't work 5.2.1

in server B I have PHP 8.2 enabled by default, result:
myphpadmin doesn't work 5.2.1

**Comment by williamdes** [comment]
🤔 
Can you send the output of `apt policy phpmyadmin` for each machine ?

**Comment by Danieleeffe1** [comment]
what command?

**Comment by Danieleeffe1** [comment]
phpmyadmin:
  Installed: 4:5.2.1+dfsg-1+focal2
  Candidate: 4:5.2.1+dfsg-1+focal2
  Version table:
 *** 4:5.2.1+dfsg-1+focal2 500
        500 http://ppa.launchpad.net/phpmyadmin/ppa/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status
     4:4.9.5+dfsg1-2ubuntu0.1~esm1 500
        500 https://esm.ubuntu.com/apps/ubuntu focal-apps-security/main amd64 Packages
     4:4.9.5+dfsg1-2 500
        500 http://ua.archive.ubuntu.com/ubuntu focal/universe amd64 Packages

**Comment by williamdes** [comment]
> did you ask this?

Yes, thank you


I edit what was said before, since you are still on Ubuntu focal you do not need PHP 8 to run phpMyAdmin.

`4:5.2.1+dfsg-1+focal2` looks up to date, that's a good starting point.

For `Unknown named parameter $$dbi` I think it's what is documented in https://github.com/phpmyadmin/phpmyadmin/wiki/DebianUbuntu#known-issues

So you will have to change a file in the file system: `/usr/share/php/Symfony/Component/DependencyInjection/ContainerBuilder.php` to apply the change: https://github.com/symfony/symfony/commit/8e34978e4e2c4bf7dac97e44e904368c4cfb54f8

Ref: https://github.com/phpmyadmin/phpmyadmin/issues/17585#issuecomment-1196750348

There is an update for Ubuntu focal but for now you need to do it the manual way I explained
Ref: https://code.launchpad.net/~williamdes/ubuntu/+source/symfony/+git/symfony/+merge/442971

**Comment by Danieleeffe1** [comment]
mmm...strange situation.

if i type the command
update-alternatives --config php
Server B has php7.4 enabled

php8.2 is highlighted in the phpmyadmin home page in the server details

it's normal?

**Comment by williamdes** [comment]
> it's normal?

I would say yes, the PHP version in the command line is defined by `update-alternatives --config php` but the one used by the webserver (apache/nginx/...) is defined in a config file

Is everything sorted out and solved for you ?

**Comment by hayhurst** [comment]
Running into the same issue...

<img width="603" alt="image" src="https://github.com/phpmyadmin/phpmyadmin/assets/15649917/dcbaf787-ee1c-466a-a657-fed3ae472abd">

**Comment by williamdes** [comment]
Did you apply https://github.com/phpmyadmin/phpmyadmin/issues/18498#issuecomment-1592981344 ?

**Comment by hayhurst** [comment]
Ah huh I see - that did the trick.

Is that fix going to get overwritten if I run any updates in the meantime? Presuming that symfony doesn't have anything in their pipeline to fix this?

**Comment by williamdes** [comment]
> Ah huh I see - that did the trick.

Cool

> Is that fix going to get overwritten if I run any updates in the meantime? Presuming that symfony doesn't have anything in their pipeline to fix this?

It's not up to symfony since they fixed it in their code
It's now up to package maintainers, obviously nobody cares except phpMyAdmin related folks

But as I said I am trying to solve this years old problem by making an update in Ubuntu: https://code.launchpad.net/~williamdes/ubuntu/+source/symfony/+git/symfony/+merge/442971

I need to do some paperwork for them to accept it.
It should overwrite the file when the package is updated on your workstation

**Comment by williamdes** [comment]
Here we go: https://bugs.launchpad.net/ubuntu/+source/symfony/+bug/2024615

🙏🏻 Let's see if we can get an Ubuntu update

**Comment by williamdes** [comment]
Duplicate of #16968
